syntax = "proto3";
package CoSiMa_rpc;

service SimulationInterface {
	rpc GetStringValue(SimString) returns (SimString) {}
	rpc SetStringValue(SimNamedString) returns (SimInt32) {}
	
	rpc DoStep(SimEmpty) returns (SimDouble) {}
	rpc SetConfig(SimConfig) returns (SimInt32) {}
}

message SimEmpty {
	//TODO find out why "google/protobuf/Empty.proto" cannot be found and use it instead
}

message SimString {
	string value = 1;
}

message SimNamedString {
	string name = 1;
	string value = 2;
}

message SimInt32 {
	int32 value = 1;
}

message SimDouble {
	double value = 1;
}

message SimConfig {
	string FMU = 1;
}
